crates/bifrost/src/providers/replicated_loglet/sequencer/mod.rs

```diff
@@ -297,7 +168,36 @@ impl<T: TransportConnect> Sequencer<T> {
             return Ok(LogletCommit::sealed());
         }
 
-        let next_write_offset = records.last_offset(self.next_write_offset)?.next();
+        if payloads.len() > self.max_in_flight_records_in_config {
+            let delta = payloads.len() - self.max_in_flight_records_in_config;
+            debug!(
+                "Resizing sequencer in-flight records capacity to allow admission for this batch. \
+                Capacity in configuration is {} and we are adding capacity of {} to it",
+                self.max_in_flight_records_in_config, delta
+            );
+            self.record_permits.add_permits(delta);
+        }
```

Comment:
Shouldn't we also update the `self.max_in_flight_records_in_config += delta` to reflect that we have increased the maximum?

Reply:
Yes, we can do that to reduce overshooting.